ADC/Core/Src/main.c

```diff
@@ -0,0 +1,184 @@
+/*
+ * 	Implement a routine that reads the internal voltage reference (ADC_IN17)
+ * 	and internal temperature sensor (ADC_IN18).
+ * 	The internal reference voltage must be in millivolts and the temperature in Celsius.
+ *
+ * 	1 - Polling		[/]
+ * 	2 - Interrupt	[ ]
+ * 	3 - DMA			[ ]
+ */
+
+#include "stm32l053xx.h"
+#include "macros.h"
+#include "init.h"
+
+uint8_t	AD_CAL 		= 0,
+		T_Delay		= 0;
+int		measure 	= 0,
+		temperature	= 0,
+		v_ref 		= 0;
+
+void ADC_Cal (void);
+void ADC_ON (void);
+void ADC_OFF (void);
+
+int main (void)
+{
+	ADC_Init();
+
+	ADC_ON();
+
+	//Reading the temperature sensor
+	if(ADC1->CR & ADC_CR_ADSTART)			//we have to be sure there's no ongoing conversion
+	{
+		ADC1->CR |= ADC_CR_ADSTP;
+		while(ADC1->CR & ADC_CR_ADSTP);
+	}
+	ADC1->CHSELR |= ADC_CHSELR_CHSEL18;		//selecting the TSEN channel
+	ADC->CCR |= ADC_CCR_TSEN; 				//enables temperature sensor
+	T_Delay = 21;							//we have to wait for 10 us for the temp sensor to wake up
+	while(T_Delay)
+	{
+		T_Delay--;
+	}
+	ADC1->CR |= ADC_CR_ADSTART;				//we start the conversion
+	T_Delay = 42;							//and wait for the properly time
+	while(T_Delay)
+	{
+		T_Delay--;
+	}
+	measure = ADC1->DR & ADC_DR_DATA;		//we store the ADC data in measure
+
+	//temp calc
+	temperature = ((measure * VDD_APPLI / VDD_CALIB) - (int32_t) *TEMP30_CAL_ADDR);
+	temperature = temperature * (int32_t)(130 - 30);
+	temperature = temperature / (int32_t)(*TEMP130_CAL_ADDR - *TEMP30_CAL_ADDR);
+	temperature = temperature + 30;
+
+	//Reading the internal reference voltage
+	if(ADC1->CR & ADC_CR_ADSTART)			//we have to be sure there's no ongoing conversion
```

Comment:
Please take a look at the Init.c archive from the TDR code and look at the ADC_config function, we use that to configure the channel we want to read at each time.